src/linear_algebra/determinant.lean

```diff
@@ -217,12 +221,53 @@ linear_map.det.map_mul f g
 lemma det_id : (linear_map.id : M →ₗ[A] M).det = 1 :=
 linear_map.det.map_one
 
-lemma det_zero {ι : Type*} [fintype ι] [nonempty ι] (b : basis ι A M) :
+/-- Multiplying a map by a scalar `c` multiplies its determinant by `c ^ dim M`. -/
+@[simp] lemma det_smul {𝕜 : Type*} [field 𝕜] {M : Type*} [add_comm_group M] [module 𝕜 M]
+  (c : 𝕜) (f : M →ₗ[𝕜] M) :
+  linear_map.det (c • f) = c ^ (finite_dimensional.finrank 𝕜 M) * linear_map.det f :=
+begin
+  by_cases H : ∃ (s : finset M), nonempty (basis s 𝕜 M),
+  { haveI : finite_dimensional 𝕜 M,
+    { rcases H with ⟨s, ⟨hs⟩⟩, exact finite_dimensional.of_finset_basis hs },
+    simp only [← det_to_matrix (finite_dimensional.fin_basis 𝕜 M), linear_equiv.map_smul,
+              fintype.card_fin, det_smul] },
+  { classical,
+    have : finite_dimensional.finrank 𝕜 M = 0 := finrank_eq_zero_of_not_exists_basis H,
+    simp [coe_det, H, this] }
+end
+
+lemma det_zero' {ι : Type*} [fintype ι] [nonempty ι] (b : basis ι A M) :
   linear_map.det (0 : M →ₗ[A] M) = 0 :=
 by { haveI := classical.dec_eq ι,
      rw [← det_to_matrix b, linear_equiv.map_zero, det_zero],
      assumption }
 
+/-- In a finite-dimensional vector space, the zero map has determinant `1` in dimension `0`,
+and `0` otherwise. -/
+@[simp] lemma det_zero {𝕜 : Type*} [field 𝕜] {M : Type*} [add_comm_group M] [module 𝕜 M] :
+  linear_map.det (0 : M →ₗ[𝕜] M) = (0 : 𝕜) ^ (finite_dimensional.finrank 𝕜 M) :=
+by simp only [← zero_smul 𝕜 (1 : M →ₗ[𝕜] M), det_smul, mul_one, monoid_hom.map_one]
+
+/-- Conjugating a linear map by a linear equiv does not change its determinant. -/
+@[simp] lemma det_conj {N : Type*} [add_comm_group N] [module A N]
+  (f : M →ₗ[A] M) (e : M ≃ₗ[A] N) :
+  linear_map.det ((e : M →ₗ[A] N).comp (f.comp e.symm)) = linear_map.det f :=
+begin
+  classical,
+  by_cases H : ∃ (s : finset M), nonempty (basis s A M),
+  { rcases H with ⟨s, ⟨b⟩⟩,
+    rw [← det_to_matrix b f, ← det_to_matrix (b.map e), to_matrix_comp (b.map e) b (b.map e),
+        to_matrix_comp (b.map e) b b, det_mul, det_mul, mul_comm, mul_assoc, ← det_mul,
+        ← to_matrix_comp],
+    simp only [linear_equiv.refl_to_linear_map, linear_equiv.trans_symm, det_one, mul_one,
+      basis.to_matrix_self, to_matrix_id_eq_basis_to_matrix, linear_equiv.comp_coe] },
```

Comment:
Rewriting with `matrix.det_conj` is somewhat more verbose, but feels more readable to me. I can also live with your current version.
```suggestion
    rw [← det_to_matrix b f, ← det_to_matrix (b.map e), to_matrix_comp (b.map e) b (b.map e),
        to_matrix_comp (b.map e) b b, ← matrix.mul_assoc, matrix.det_conj],
    { rw [← to_matrix_comp, linear_equiv.comp_coe, e.symm_trans,
          linear_equiv.refl_to_linear_map, to_matrix_id] },
    { rw [← to_matrix_comp, linear_equiv.comp_coe, e.trans_symm,
          linear_equiv.refl_to_linear_map, to_matrix_id] } },
```